packages/react-native-bpk-component-button-link/src/common-types.js

```diff
@@ -21,6 +21,7 @@
 import { type Node } from 'react';
 import PropTypes from 'prop-types';
 import { makeThemePropType } from 'react-native-bpk-theming';
+import type { PressEvent } from 'react-native/Libraries/Types/CoreEventTypes';
```

Comment:
Are you meant to use stuff like this? Isn't there a danger that they change what's exposed as it's not part of the official API?

Reply:
There is that risk, but I don't see another way of doing it as it doesn't appear to be exported on the public interface, and if it breaks we'll find out as soon as we try to upgrade RN and can change it.

I did try this:

```
type TouchableProps = ElementProps<typeof TouchableWithoutFeedback>;
type TouchableOnPressProp = $PropertyType<TouchableProps, 'onPress'>;
```

But I get a Flow error and I couldn't figure it out:

```
Cannot instantiate $PropertyType because property onPress is missing in mixed [1].

     29│ } from './customPropTypes';
     30│
     31│ type TouchableProps = ElementProps<typeof TouchableWithoutFeedback>;
 [1] 32│ type TouchableOnPressProp = $PropertyType<TouchableProps, 'onPress'>;
```

@matthewdavidson may have an opinion as he's done similar things in his Flow-ing PRs.

Reply:
You can use `SyntheticEvent<>`, no?

Reply:
It doesn't like that anymore, that's what we were using before this PR:

![Screen Shot 2019-03-14 at 14 33 34](https://user-images.githubusercontent.com/73652/54365126-2f8eb180-4666-11e9-9e62-2ea820445b45.png)